### PR TITLE
Adding Recursive option.

### PR DESCRIPTION
**Pull request by vinaykapadia**

### CODE REVIEW

File: src/Sarif.Multitool/MergeCommand.cs [hunk 1]
```diff
@@ -74,7 +77,7 @@ private static IEnumerable<string> GetSarifFiles(IEnumerable<string> files)
 				    directory = Path.GetDirectoryName(path) ?? path;
 				    filename = Path.GetFileName(path) ?? "*";
 			    }
-			    foreach (var file in Directory.GetFiles(directory, filename, SearchOption.AllDirectories))
+			    foreach (var file in Directory.GetFiles(directory, filename, searchOption))
 			    {
 				    if (file.EndsWith(".sarif", StringComparison.InvariantCultureIgnoreCase))
```
Comment (michaelcfanning) [inline src/Sarif.Multitool/MergeCommand.cs]: >InvariantCultureIgnoreCase [](start = 53, length = 26)

Use OrdinalIgnoreCase here instead

File: src/Sarif.Multitool/MergeCommand.cs [hunk 2]
```diff
@@ -74,7 +77,7 @@ private static IEnumerable<string> GetSarifFiles(IEnumerable<string> files)
 				    directory = Path.GetDirectoryName(path) ?? path;
 				    filename = Path.GetFileName(path) ?? "*";
 			    }
-			    foreach (var file in Directory.GetFiles(directory, filename, SearchOption.AllDirectories))
+			    foreach (var file in Directory.GetFiles(directory, filename, searchOption))
```
Comment (michaelcfanning) [inline src/Sarif.Multitool/MergeCommand.cs]: >var [](start = 16, length = 3)

make 'string' type explicit. standard says don't use var to capture assignment from any function call , as type isn't explicit in code

File: src/Sarif.Multitool/MergeCommand.cs [hunk 3]
```diff
@@ -59,9 +58,13 @@ private static IEnumerable<Run> GetAllRuns(IEnumerable<string> sarifFiles)
...
 
-	    private static IEnumerable<string> GetSarifFiles(IEnumerable<string> files)
+	    private static IEnumerable<string> GetSarifFiles(MergeOptions mergeOptions)
 	    {
-		    foreach (var path in files)
+		    var searchOption = mergeOptions.Recursive 
+				? SearchOption.AllDirectories 
+				: SearchOption.TopDirectoryOnly;
+
+		    foreach (var path in mergeOptions.Files)
```
Comment (michaelcfanning) [inline src/Sarif.Multitool/MergeCommand.cs]: >var [](start = 15, length = 3)

same note, use 'string' not var here

File: src/Sarif.Multitool/MergeOptions.cs [hunk 4]
```diff
@@ -14,5 +14,12 @@ internal class MergeOptions : MultitoolOptionsBase
             HelpText = "Files to process (wildcards ? and * allowed).",
             Required = false)]
         public IList<string> Files { get; internal set; }
-    }
+
+	    [Option(
+		    'r',
+		    "recursive",
```
Comment (michaelcfanning) [inline src/Sarif.Multitool/MergeOptions.cs]: >recursive [](start = 7, length = 9)

use 'recurse' as option and property name

File: src/Sarif.Multitool/MergeCommand.cs (hunk 3 above)
Comment (vinaykapadia) [reply src/Sarif.Multitool/MergeCommand.cs]: Done.

---
In reply to: [141649387](https://github.com/Microsoft/sarif-sdk/pull/627#discussion_r141649387) [](ancestors = 141649387)

File: src/Sarif.Multitool/MergeCommand.cs (hunk 2 above)
Comment (vinaykapadia) [reply src/Sarif.Multitool/MergeCommand.cs]: Done.

---
In reply to: [141649329](https://github.com/Microsoft/sarif-sdk/pull/627#discussion_r141649329) [](ancestors = 141649329)

File: src/Sarif.Multitool/MergeCommand.cs (hunk 1 above)
Comment (vinaykapadia) [reply src/Sarif.Multitool/MergeCommand.cs]: Done.

---
In reply to: [141649198](https://github.com/Microsoft/sarif-sdk/pull/627#discussion_r141649198) [](ancestors = 141649198)

File: src/Sarif.Multitool/MergeOptions.cs (hunk 4 above)
Comment (vinaykapadia) [reply src/Sarif.Multitool/MergeOptions.cs]: Done.

---
In reply to: [141649490](https://github.com/Microsoft/sarif-sdk/pull/627#discussion_r141649490) [](ancestors = 141649490)

**Pull request merged**